crates/matrix-sdk-crypto/src/store/error.rs

```diff
@@ -31,6 +31,10 @@ pub enum CryptoStoreError {
     #[error("can't save/load sessions or group sessions in the store before an account is stored")]
     AccountUnset,
 
+    /// An account was saved but no own device was found.
+    #[error("Incomplete account, account data was saved but no own device was found")]
+    IncompleteAccountNoOwnDevice,
```

Comment:
I'm not enthusiastic about adding this here as yet another error code that any API can return: error handling is really hard when you call matrix-sdk-crypto APIs because there are thousands of possible error cases.

Maybe we can specialise the result from `with_store`?

Reply:
Yes I was not sure about this one. What do you mean by "specialise" the result?

Reply:
make it return a different error type in its Result, which more closely reflects the actual possible failure modes of that particular method.